gegelatilib/include/tpg/tpgGraph.h

```diff
@@ -23,6 +23,39 @@ namespace TPG {
 		*/
 		TPGGraph(const Environment& e) : env{ e } {};
 
+		/**
+		* \brief delete copy constructor
+		*/
+		TPGGraph(const TPGGraph & model) = delete;
+
+		/**
+		* \brief TPGGraph move assignment operator
+		*
+		* \param[in] model the TPGGraph to copy
+		*/
+		TPGGraph(TPGGraph && model) noexcept : env{model.getEnvironment()}
+		{
+			swap(*this, model);
+		}
+
+		/**
+		*	\brief Helper function for move constructor.
+		*
+		*	Swaps the TPGGraphs objects.
+		*/
+		friend inline void swap(TPGGraph &a, TPGGraph &b)
+		{
+			using std::swap;
+			swap(a.vertices, b.vertices);
+			swap(a.edges, b.edges);
+		}
+
+
+		/**
+		*	\brief assignement operator for class TPGGraph
+		*/
+		TPGGraph & operator=(TPGGraph model);
```

Comment:
I trust you on this.. I'm not fluent in move constructor.

If you're not 100% sure, please check with Farresti if this is the right thing to do :).

Reply:
That's already his tip  ;-)